tests/conftest.py

```diff
@@ -111,10 +111,12 @@ def factory(source):
         assert source
 
         # Now we need to adjust indentation. Compute how much the first line of
-        # the body is indented by, then dedent all lines by that amount.
+        # the body is indented by, then dedent all lines by that amount. Blank
+        # lines don't matter indentation-wise, and might not be indented to begin
+        # with, so just replace them with a simple newline.
         line = source[0]
         indent = len(line) - len(line.lstrip())
-        source = [line[indent:] for line in source]
+        source = [line[indent:] if line.strip() else '\n' for line in source]
```

Comment:
Without this change, we were basically stripping out blank lines from `@pyfile` code, which made line numbers hard to predict. It doesn't matter as much when using `#@` markers, but it can still be confusing in the logs.